commons/src/main/java/org/infinispan/counter/api/StrongCounter.java

```diff
@@ -0,0 +1,94 @@
+package org.infinispan.counter.api;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * The strong consistent counter interface.
+ * <p>
+ * It provides atomic updates for the counter. All the operations are perform asynchronously and they complete the
+ * {@link CompletableFuture} when completed.
+ * <p>
+ * The implementation may support weakly consistent reads via {@link #weakGetValue()}.
+ *
+ * @author Pedro Ruivo
+ * @since 9.0
+ */
+public interface StrongCounter {
+
+   /**
+    * @return The counter name.
+    */
+   String getName();
+
+
+   /**
+    * It fetches the current value.
+    * <p>
+    * It may go remotely to fetch the current value.
+    *
+    * @return The current value.
+    */
+   CompletableFuture<Long> getValue();
+
+   /**
+    * Returns the locally cached value.
+    * <p>
+    * This value may not up-to-date and it may no have the most recent updates from local and remote nodes.
+    *
+    * @return The locally cached value.
+    */
+   default long weakGetValue() {
```

Comment:
I think that it should be removed from Strong counters, and Weak counters should rely purely on those clustered listeners.